juntagrico/views_subscription.py

```diff
@@ -306,11 +307,23 @@ def activate_subscription(request, subscription_id):
     change_date = request.session.get('changedate', None)
     try:
         subscription.activate(change_date)
+        add_subscription_member_to_activity_area(subscription)
     except ValidationError as e:
         return error_page(request, e.message)
     return return_to_previous_location(request)
 
 
+def add_subscription_member_to_activity_area(subscription):
+    activity_area = ActivityArea.objects.filter(auto_add_new_members=True)
+    member_ids = SubscriptionMembership.objects.filter(subscription=subscription).values_list('member')
+    members = Member.objects.filter(pk__in=member_ids)
+
+    for member in members:
+        for area in activity_area:
+            if member not in area.members.all():
+                area.members.add(member)
+
```

Comment:
[area.members.add(*members) for area in activity_area]
though not sure if *members works eventually you need to convert members to a list

Reply:
done, works